.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,60 @@
+name: Integration Tests
```

Comment:
It seems like this should just replace the existing `UI tests` at https://github.com/PrefectHQ/prefect/blob/main/.github/workflows/ui-tests.yml

It'd be nice to continue to include the "UI" prefix on the file and workflow name for clarity.

Reply:
Yes we probably don't need that ui-tests workflow once this is in main as this workflow will fail if it cannot build the ui. 

As far as using the "UI" prefix I'm not sure. This does use the actual api. Which means a test could be broken because something is wrong with the API.

Reply:
It's not actually going to run any tests defined outside of the UI code though. I'd expect integration tests to cover the full workflow from Python -> API -> UI 🤷 Perhaps we'll add additional integration-style tests to this workflow in the future though.

Reply:
That's my plan. I'm hoping to leverage @peytonrunyan 's work [here](https://github.com/PrefectHQ/automated-qa) so that when the tests run there is data from the python client visible in the UI and we can assert on that.

Reply:
This is way more up-to-date fwiw: https://github.com/PrefectHQ/prefect/pull/6132

I'm getting my windows VM setup to test it, and after that it just needs a review.